monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py

```diff
@@ -15,8 +29,37 @@ def __init__(
         self._agent_id = agent_id
         self._agent_event_publisher = agent_event_publisher
 
+        self._send_bitcoin_mining_request_periodically = PeriodicCaller(
+            callback=self._send_bitcoin_mining_request,
+            period=REQUEST_INTERVAL,
+            name="Cryptojacking.BitcoinMiningNetworkTrafficSimulator",
+        )
+
     def start(self):
-        pass
+        logger.info("Starting Bitcoin mining network traffic simulator")
+
+        self._send_bitcoin_mining_request_periodically.start()
+
+    @handle_island_errors
```

Comment:
Aside from the fact that we shouldn't couple these components, I don't even think this decorator will work. The http client uses requests, whereas this component is using urllib.

On a separate note, why urllib and not requests? It's fine to leave it as is. Just wondering. For consistency we may want to switch to requests, but up to you.

Reply:
I just picked it up from the script I used for testing when we did the spike, planned on switching to `requests` later but it slipped my mind.